### DISCUSSION TITLE
ValueError: [E002] while installing 'sentiws'

### DISCUSSION THREAD

**Discussion by syon1203** [explosion/spaCy#10698] [Help: Installation] [answered]
Hello, I'm trying to use 'sentiws' package which is for analysing sentiments in German. Sadly there's an error while installation. I already installed transformer and tried other suggestions written for similar problems. Does anyone knows how to fix it or had same problem before? 

[https://spacy.io/universe/project/spacy-sentiws](url)
```
import spacy
from spacy_sentiws import spaCySentiWS

nlp = spacy.load('de_core_news_sm')
nlp.add_pipe('sentiws', config={'sentiws_path': 'data/sentiws'})
doc = nlp('Die Dummheit der Unterwerfung blüht in hübschen Farben.')

for token in doc:
    print('{}, {}, {}'.format(token.text, token._.sentiws, token.pos_))
```

> 
> ValueError: [E002] Can't find factory for 'sentiws' for language German (de). This usually happens when spaCy calls `nlp.create_pipe` with a custom component name that's not registered on the current language class. If you're using a Transformer, make sure to install 'spacy-transformers'. If you're using a custom component, make sure you've added the decorator `@Language.component` (for function components) or `@Language.factory` (for class components).
> 
> Available factories: attribute_ruler, tok2vec, merge_noun_chunks, merge_entities, merge_subtokens, token_splitter, doc_cleaner, parser, beam_parser, entity_linker, ner, beam_ner, entity_ruler, lemmatizer, tagger, morphologizer, senter, sentencizer, textcat, spancat, textcat_multilabel, transformer
> 

spaCy version    3.2.4 
Python version   3.10.2                        
Pipelines        de_core_news_sm (3.2.0)

**Comment by thomashacker** [comment, marked as answer]
Hello,
Have you installed `spacy_sentiws` with `pip` ? I think the problem is that the owner of the repository hasn't updated the package release yet which means that `pip install spacy_sentiws` installs an older version that is only compatible with `spaCy v2`. The solution would be to install the package from source (https://github.com/Liebeck/spacy-sentiws)

**Reply by syon1203** [reply]
Thanks for your help!  It was a version problem and solution worked out.
